src/main/java/com/amazon/ion/impl/IonReaderTextRawX.java

```diff
@@ -126,7 +117,11 @@ private final String get_state_name(int state) {
     static final int ACTION_FINISH_LOB           = 13;
     static final int ACTION_FINISH_DATAGRAM      = 14;
     static final int ACTION_EOF                  = 15;
-    static final int ACTION_count                = 16;
+
+    static final int ACTION_START_E_EXPRESSION = 16;
+    static final int ACTION_START_EXPRESSION_GROUP = 17;
+    static final int ACTION_TDL_EXPRESSION_OR_OPERATOR = 18;
```

Comment:
Why is this needed? TDL is written using plain-old-ion, and so the Macro Compiler should be able to read it without needing any special tokens, right?

Reply:
Yes, it can read it. The action is needed so that the text parser can properly *classify* the token as either an expression or a regular operator.

Reply:
I'll look into just using the MacroCompiler for this classification in TDL.